File: backend/app/schemas/prazo.py
import uuid
from datetime import date

from pydantic import BaseModel, ConfigDict

from app.models.enums import StatusPrazo, TipoPrazo


class PrazoBase(BaseModel):
    data_prazo: date
    tipo: TipoPrazo
    descricao: str | None = None
    responsavel_id: uuid.UUID | None = None


class PrazoCreate(PrazoBase):
    pass


class PrazoUpdate(BaseModel):
    data_prazo: date | None = None
    tipo: TipoPrazo | None = None
    status: StatusPrazo | None = None
    descricao: str | None = None
    responsavel_id: uuid.UUID | None = None


class ProcessoResumo(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    numero_cnj: str


class PrazoRead(PrazoBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    processo_id: uuid.UUID
    status: StatusPrazo
    movimentacao_origem_id: uuid.UUID | None
    processo: ProcessoResumo | None = None


class PrazoListResponse(BaseModel):
    items: list[PrazoRead]
    total: int
    page: int
    page_size: int


class DashboardPrazos(BaseModel):
    inicio: date
    fim: date
    total_pendentes: int
    vencidos: int
    prazos: list[PrazoRead]

File: frontend/src/types/prazo.ts
export type TipoPrazo = "peremptorio" | "dilatorio" | "recursal";
export type StatusPrazo = "pendente" | "cumprido" | "perdido";

export interface ProcessoResumo {
  id: string;
  numero_cnj: string;
}

export interface Prazo {
  id: string;
  processo_id: string;
  data_prazo: string;
  tipo: TipoPrazo;
  status: StatusPrazo;
  descricao: string | null;
  responsavel_id: string | null;
  movimentacao_origem_id: string | null;
  processo?: ProcessoResumo | null;
}

export interface PrazoListResponse {
  items: Prazo[];
  total: number;
  page: number;
  page_size: number;
}

export interface DashboardPrazos {
  inicio: string;
  fim: string;
  total_pendentes: number;
  vencidos: number;
  prazos: Prazo[];
}

export interface PrazoCreateInput {
  data_prazo: string;
  tipo: TipoPrazo;
  descricao?: string;
  responsavel_id?: string;
}

export interface PrazoUpdateInput {
  data_prazo?: string;
  tipo?: TipoPrazo;
  status?: StatusPrazo;
  descricao?: string;
}
